osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -453,8 +462,32 @@ protected override void PopIn()
                       .MoveToY(0, duration, Easing.OutQuint)
                       .FadeIn(duration, Easing.OutQuint);
 
-                if (!allFiltered)
-                    nonFilteredColumnCount += 1;
+                if (allFiltered)
+                    continue;
+
+                int columnNumber = nonFilteredColumnCount;
+                Scheduler.AddDelayed(() =>
+                {
+                    var channel = columnAppearSample?.GetChannel();
+                    if (channel == null) return;
+
+                    // don't play sample if column isn't visible, but try to play it at least 5 times (assuming there's at least 5 visible columns in `columnFlow`)
```

Comment:
This conditional is a tad weird to read. To confirm, the goal is to play the sound even for off-screen columns up to a max of 5 to make the effect sound correct?

Reply:
yes

Reply:
Hmm, the only discrepancy is that it will play 6 times with the code you have.